db/db_impl/db_impl.cc

```diff
@@ -4331,6 +4331,24 @@ void DBImpl::ReleaseSnapshot(const Snapshot* s) {
       }
       bottommost_files_mark_threshold_ = new_bottommost_files_mark_threshold;
     }
+
+    // Avoid to go through every column family by checking a global threshold
+    // first.
+    if (oldest_snapshot >= standalone_range_deletion_files_mark_threshold_) {
```

Comment:
If I'm reading the code correctly, this is only going to be true when there's an eligible range del file? I believe the potential hot path CPU performance concerns (such as more work on *every* release of the oldest snapshot) have been taken care of.